From 15f39102145af20800d906843e4dd7f8f7340245 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 11 Dec 2013 20:41:18 +0100
Subject: [PATCH 321/327] MINOR: checks: fix agent check interval computation

start_checks() only used to consider the health checks intervals to
compute the start interval, so if an agent had a faster check than
all health checks, it would be significantly delayed.
---
 src/checks.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/checks.c b/src/checks.c
index ec05781..9d906bd 100644
--- a/src/checks.c
+++ b/src/checks.c
@@ -1771,6 +1771,11 @@ int start_checks() {
 			    (!mininter || mininter > srv_getinter(&s->check)))
 				mininter = srv_getinter(&s->check);
 
+			if ((s->agent.state & CHK_ST_CONFIGURED) &&
+			    (srv_getinter(&s->agent) >= SRV_CHK_INTER_THRES) &&
+			    (!mininter || mininter > srv_getinter(&s->agent)))
+				mininter = srv_getinter(&s->agent);
+
 			nbcheck++;
 		}
 	}
-- 
1.8.1.5

